server/src/main/java/com/testsigma/service/TestStepService.java

```diff
@@ -574,8 +577,40 @@ private List<ActionTestDataMap> getMapsList(){
 
         return actionsMap;
     }
+    public void bulkDelete(Long[] ids) throws ResourceNotFoundException {
+        for (Long id:ids){
+              this.destroy(this.find(id));
+            }
+    }
 
-
+    private Boolean checkBulkDeleteStepsHaveAllPrerequisiteSteps(List<Long> ids,List<TestStep> testStepList){
+        for (TestStep testStep:testStepList){
+            if(!ids.contains(testStep.getId())){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public List<TestStepDTO> checkForLinkedTestCases(Long[] ids) throws ResourceNotFoundException {
+        Long testCaseId = this.find(ids[0]).getTestCaseId();
+        List<TestStepDTO> testSteps = new ArrayList<>();
+        List<Long> idsList=Arrays.asList(ids);
+        int index=0;
```

Comment:
This index is not required. Remove this. You can simply add an item to the arraylist without the index